erase_stress_example/main/erase_stress_example.c

```diff
@@ -0,0 +1,150 @@
+
+    FILE *f;
+
+    // read from file only on consecutive runs where the file is already created and written to
+    // (covers the very first run of this example)
+    if (full_runs != 0) {
+        // Open file for reading
+        ESP_LOGI(TAG, "Reading file");
+        f = fopen("/spiflash/hello.txt", "rb");
+        if (f == NULL) {
+            ESP_LOGE(TAG, "Failed to open file for reading");
+            return;
+        }
+        char line[128];
+        fgets(line, sizeof(line), f);
+        fclose(f);
+        // strip newline
+        char *pos = strchr(line, '\n');
+        if (pos) {
+            *pos = '\0';
+        }
+        fclose(f);
+        ESP_LOGI(TAG, "Read from file: '%s'", line);
+    }
+
+    // write to file, creating if it does not exist
+    ESP_LOGI(TAG, "Opening file");
+    f = fopen("/spiflash/hello.txt", "wb");
+    if (f == NULL) {
+        ESP_LOGE(TAG, "Failed to open file for writing");
+        return;
+    }
+    fprintf(f, "written using ESP-IDF %s\n", esp_get_idf_version());
+    fclose(f);
+    ESP_LOGI(TAG, "File written");
+
+    // Unmount FATFS
+    ESP_LOGI(TAG, "Unmounting FAT filesystem");
+    ESP_ERROR_CHECK( esp_vfs_fat_spiflash_unmount_rw_wl(base_path, s_wl_handle));
+
+#if !FAT_ONLY
+    /*
+     * Part II: erase stressing
+     */
+
+    // get wear leveled data partition
+    const esp_partition_t *storage;
+    storage = esp_partition_find_first(ESP_PARTITION_TYPE_DATA, ESP_PARTITION_SUBTYPE_ANY, "storage");
+    if (!storage) {
+        ESP_LOGE(TAG, "Failed to find first data partition labeled 'storage'");
+        return;
+    }
+
+    ESP_ERROR_CHECK(wl_mount(storage, &s_wl_handle));
+
+    // perform iterations of erasing a single sector to artificially stress the memory
+    #define ERASE_ITERATIONS (16*50)
+
+    for (int i = 0; i < ERASE_ITERATIONS; i++) {
+        if (i % 100 == 0) {
+            ESP_LOGI(TAG, "%s: erase stressing iteration %u/%u", __func__, i, ERASE_ITERATIONS);
+
+            uint32_t rand = esp_random();
+
+            // random early break allows for some variance in erase counts
+            if (rand % 100 < 5) {
+                ESP_LOGI(TAG, "%s: random cycle break at iteration %u", __func__, i);
+                break;
+            }
+
+            // or remounting will move dummy sector 'a lot', making move_count increment (and in turn mapping shift) occur sooner
+            if (rand % 100 < 20) {
+                ESP_LOGI(TAG, "%s: will remount cycle at iteration %u", __func__, i);
+                for (int j = 0; j < (rand % 100 + 15); j++) {
+                    // unmounting always moves dummy sector
+                    ESP_ERROR_CHECK(wl_unmount(s_wl_handle));
+                    ESP_ERROR_CHECK(wl_mount(storage, &s_wl_handle));
+                }
+            }
+            // erase a sector around the middle of wear leveled partition
+            wl_erase_range(s_wl_handle, wl_size(s_wl_handle)/2 + (i%10 + rand%10), CONFIG_WL_SECTOR_SIZE);
```

Comment:
missing retval check. general note for whole code (wherever applicable)

Reply:
yoinked a result check macro from WL which now checks wl erases/mounts/unmounts